test/recovery/random-abort.c

```diff
@@ -317,12 +318,23 @@ main(int argc, char *argv[])
 		 * in the table after recovery.  Since we did write-no-sync, we
 		 * expect every key to have been recovered.
 		 */
-		for (;; ++count) {
+		for (last_key = UINT64_MAX;; ++count, last_key = key) {
 			ret = fscanf(fp, "%" SCNu64 "\n", &key);
 			if (ret != EOF && ret != 1)
 				testutil_die(errno, "fscanf");
 			if (ret == EOF)
 				break;
+			/*
+			 * If we're unlucky, the last line may be a partially
+			 * written key at the end that can result in a false
+			 * negative error for a missing record.  Detect it.
+			 */
+			if (last_key != UINT64_MAX && key != last_key + 1) {
+				printf("%s: Ignore partial record %" PRIu64
+				    " last valid key %" PRIu64 "\n",
+				    fname, key, last_key);
+				break;
+			}
```

Comment:
I coded this up and I am going to leave it as-is.  Testing with `fgets` and looking for the newline runs much slower than the `fscanf` and the key check.  Running 3 times on the failed data, the `fscanf` took between 2.1-2.5 seconds per rn.  The `fgets` took 20-33 seconds per run.  This is test code, and that is sufficient reason.